### ISSUE TITLE
[Bug]: UPSERT doesn't work if there is a calculated column - Mysql

### ISSUE THREAD

**Issue by ayush3797** [Azure/data-api-builder#1583]
On similar lines to option A discussed for MsSql in https://github.com/Azure/data-api-builder/issues/1453, need to implement a similar solution for MySql.

**Comment by ayush3797** [comment]
**For tables in MySql**, _generated columns (also known as computed columns)_ are not updatable. Refer here: [Generated columns in MySql](https://dev.mysql.com/doc/refman/5.7/en/create-table-generated-columns.html#:~:text=CREATE%20VIEW%20v1%20AS%20SELECT%20%2A%2C%20CONCAT%20%28first_name%2C%27,AS%20full_name%20FROM%20t1%3B%20SELECT%20full_name%20FROM%20v1%3B) to get more idea about generated columns. Generated columns can be VIRTUAL/STORED. 

- VIRTUAL: Column values are not stored, but are evaluated when rows are read, immediately after any BEFORE triggers. A virtual column takes no storage.
- STORED: Column values are evaluated and stored when rows are inserted or updated. A stored column does require storage space and can be indexed.

I came across two solutions to collect metadata about generate columns:

1. Use the below sql:
```sql
SELECT COLUMN_NAME,EXTRA
FROM INFORMATION_SCHEMA.COLUMNS
WHERE TABLE_SCHEMA = 'mysql'
  AND TABLE_NAME = 'todos'
  AND COLUMN_DEFAULT IS NULL
  AND EXTRA = 'VIRTUAL GENERATED';
```
But this solution only works for virtual generated columns and not the stored ones.

2. Use the below sql to get all the generated columns which have generation_expression != NULL.
```sql
select column_name,generation_expression from information_schema.columns 
where table_schema='mysql' and table_name = 'todos';
```
This solution works for both STORED/VIRTUAL generated columns. Will go ahead with this.

**Comment by ayush3797** [comment]
While I was working on implementing this solution for MySql, I encountered a limitation of MySql. Currently if you see, upsert queries for MySql is generated as:

```sql
SET @LU_0 := 0;
SET @LU_1 := 0;
SET @LU_2 := 0;
SET @LU_3 := 0;
SET @LU_4 := 0;
UPDATE `sales` SET `sales`.`item_name` = @param1, `sales`.`subtotal` = @param2, `sales`.`tax` = @param3 , `id` = (SELECT @LU_0 := `id`), `item_name` = (SELECT @LU_1 := `item_name`), `subtotal` = (SELECT @LU_2 := `subtotal`), `tax` = (SELECT @LU_3 := `tax`), `total` = (SELECT @LU_4 := `total`) WHERE `sales`.`id` = @param0;  SET @ROWCOUNT=ROW_COUNT(); SELECT @LU_0 AS `id`, @LU_1 AS `item_name`, @LU_2 AS `subtotal`, @LU_3 AS `tax`, @LU_4 AS `total` WHERE @ROWCOUNT > 0;
``` 

Now here, assume that we know that the column `total = subtotal+tax` is a computed column . Obviously, since total is a computed column, we cannot update its value and hence cannot include it in the update clause.But with MySql, we have a limitation that we can only include those columns in the select statement which were updated via the update statement. If we omit the part `total` = (SELECT @LU_4 := `total`)` from this query, we will get back the value for `total` as `null`. It is to be noted in the database table though, the correct value of total would be updated. Its just that we would get null because of the above limitation.


Possible solutions:

1. Maybe if instead of clubbing the update and select query together, an update followed by a select might help?

For now, will skip implementing the solution for MySql.
